src/main/java/net/minestom/server/entity/Entity.java

```diff
@@ -181,17 +156,18 @@ public static Entity getEntity(int id) {
     }
 
     /**
-     * Gets an entity based on its UUID (from {@link #getUuid()}).
+     * Gets an entity based on its uuid (from {@link #getUuid()} ()}).
+     * <p>
+     * Entity id are unique server-wide.
      *
-     * @param uuid the entity UUID
-     * @return the entity having the specified uuid, null if not found
+     * @param uuid the entity unique universal id
+     * @return the entity having the specified id, null if not found
      */
     @Nullable
-    public static Entity getEntity(@NotNull UUID uuid) {
-        return Entity.entityByUuid.getOrDefault(uuid, null);
+    public static Entity getEntity(UUID uuid) {
+        return Entity.entityById.getOrDefault((int) uuid.getLeastSignificantBits(), null);
```

Comment:
I am personally still not ok with this, it does add a bit of confusion by giving Entity and Player a different meaning of UUID

Reply:
That really _is_ the point -- An Entity's UUID is **not** the same as a Player's UUID.

Reply:
What is the difference? Without it you cannot save any entity persistently, cannot change UUID after initialization, and `Entity#getEntity(UUID)` will not work for players